File: Spring Security Tutorials/second/src/main/java/com/example/second/repositories/UserRepository.java
package com.example.second.repositories;

import com.example.second.entities.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;

import java.util.Optional;

public interface UserRepository extends JpaRepository<User, Integer> {

    @Query("select u from User u where u.username = :username") // JPQL syntax
//    @Query(value = "select * from users where username = :username", nativeQuery = true) // native SQL
    Optional<User> findUserByUsername(String username);
}

File: Project/IDM/src/main/kotlin/com/project/idm/persistence/repositories/UserRepository.kt
package com.project.idm.persistence.repositories

import com.project.idm.data.entities.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import java.util.*

interface UserRepository : JpaRepository<User, Int> {

    @Query("select u from User u where u.username = :username") // JPQL syntax
    fun findUserByUsername(username: String): Optional<User>
}